springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/AsyncPublisherAnnotationScanner.java

```diff
@@ -0,0 +1,88 @@
+package io.github.stavshamir.springwolf.asyncapi.scanners.channels.operationdata.annotation;
+
+import com.asyncapi.v2.binding.OperationBinding;
+import io.github.stavshamir.springwolf.asyncapi.scanners.channels.ChannelPriority;
+import io.github.stavshamir.springwolf.asyncapi.scanners.channels.annotation.SpringPayloadAnnotationTypeExtractor;
+import io.github.stavshamir.springwolf.asyncapi.scanners.channels.operationdata.AbstractOperationDataScanner;
+import io.github.stavshamir.springwolf.asyncapi.scanners.classes.ComponentClassScanner;
+import io.github.stavshamir.springwolf.asyncapi.types.OperationData;
+import io.github.stavshamir.springwolf.asyncapi.types.ProducerData;
+import io.github.stavshamir.springwolf.schemas.SchemasService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.EmbeddedValueResolverAware;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringValueResolver;
+
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toSet;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+@Order(value = ChannelPriority.ASYNC_ANNOTATION)
+public class AsyncPublisherAnnotationScanner extends AbstractOperationDataScanner implements EmbeddedValueResolverAware {
+    private StringValueResolver resolver;
+    private final ComponentClassScanner componentClassScanner;
+    private final SchemasService schemasService;
+
+    private final List<OperationBindingProcessor> operationBindingProcessors;
+
+    @Override
+    public void setEmbeddedValueResolver(StringValueResolver resolver) {
+        this.resolver = resolver;
+    }
+
+    @Override
+    protected SchemasService getSchemaService() {
+        return this.schemasService;
+    }
+
+    @Override
+    protected List<OperationData> getOperationData() {
+        return componentClassScanner.scan().stream()
+                .map(this::getAnnotatedMethods)
+                .flatMap(Collection::stream)
+                .map(this::mapMethodToOperationData)
+                .collect(Collectors.toList());
+    }
+
+    private Set<Method> getAnnotatedMethods(Class<?> type) {
+        Class<AsyncPublisher> annotationClass = AsyncPublisher.class;
+        log.debug("Scanning class \"{}\" for @\"{}\" annotated methods", type.getName(), annotationClass.getName());
+
+        return Arrays.stream(type.getDeclaredMethods())
+                .filter(method -> method.isAnnotationPresent(annotationClass))
+                .collect(toSet());
+    }
+
+    private OperationData mapMethodToOperationData(Method method) {
+        log.debug("Mapping method \"{}\" to channels", method.getName());
+
+        Map<String, OperationBinding> operationBindings = AsyncAnnotationScannerUtil.processBindingFromAnnotation(method, operationBindingProcessors);
```

Comment:
I might be missing something, but what happens if not operation binding annotations has been used?

Reply:
If no operation binding is used, also none is added.
No exception is thrown.

It is also possible to add multiple bindings to the same method (amqp + kafka) - if that makes sense in the application

Reply:
Makes sense, good design